app/javascript/react/components/Map/ControlPanel/ControlPanel.tsx

```diff
@@ -1,46 +1,60 @@
-import * as React from "react";
-import { useState, useEffect } from "react";
-import { useSelector } from "react-redux";
+import React, { useEffect, useRef, useState } from "react";
 import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
 
-import * as S from "./ControlPanel.style";
-import { setMapConfigId, setMapTypeId } from "../../../store/mapSlice";
-import { useAppDispatch } from "../../../store/hooks";
-import { RootState } from "../../../store";
 import { MapTypeId, ViewMode } from "../../../types/map";
+import { UrlParamsTypes, useMapParams } from "../../../utils/mapParamsHandler";
+import * as S from "./ControlPanel.style";
 
 const ControlPanel: React.FC = () => {
-  const dispatch = useAppDispatch();
-  const mapConfigId = useSelector((state: RootState) => state.map.mapConfigId);
+  const { mapTypeId, searchParams } = useMapParams();
+  const navigate = useNavigate();
+  const isFirstRender = useRef(true);
   const { t } = useTranslation();
 
   const [viewMode, setViewMode] = useState<ViewMode>(ViewMode.MAP);
   const [isTerrainChecked, setIsTerrainChecked] = useState<boolean>(false);
   const [isLabelsChecked, setIsLabelsChecked] = useState<boolean>(false);
 
+  const updateURLParams = (param: string, value: string) => {
+    const newSearchParams = new URLSearchParams(searchParams);
+    newSearchParams.set(param, value);
+    navigate({ search: newSearchParams.toString() }, { replace: true });
+  };
```

Comment:
I'd take it event further and create a custom hook for updating URL params that can be used across other components (i.e. Map component). Also you can wrap this function in Callback relying on search params and navigate to avoid unnecessary re-renders. WDYT?